pysces/body.py

```diff
@@ -48,11 +49,78 @@ def cylinder(radius, num_points):
     return Body(points)
 
 def flat_plate(num_points):
-    """Return a flat plate with the given number of points"""
-    x = np.linspace(1, 0, num_points)
+    """
+    Return a flat plate with the given number of points.  In body coordinates
+    the plate runs from (0,0) to (1,0)."""
+    x = np.linspace(1, 0, num_points) # from 1 to 0 so trailing edge at index 0
     y = np.zeros_like(x)
     return Body(np.array([x, y]).T)
 
+def joukowski_foil(xcenter=-.1, ycenter=.1, a=1, numpoints=32):
+    """
+    Return a Joukowski foil centered with the trailing edge at (2a,0), and 
+    with a total of numpoints along the boundary.
+
+    (xcenter,ycenter) and a are the center and radius of the preimage circle.
+    Note that xcenter should be negative and small.  Its magnitude determines
+    the bluffness of the foil.  Meanwhile, ycenter should be small.  It
+    determines the magnitude of the camber; positive gives upward
+    camber, and negative gives downward camber.
+
+    (See sections 4.6 - 4.9 of Acheson.)"""
+    t = np.linspace(0,2*np.pi,numpoints)
+    r = np.sqrt((a-xcenter)**2+ycenter**2)
+    chi = xcenter + r*np.cos(t)
+    eta = ycenter + r*np.sin(t)
+    mag2 = chi*chi + eta*eta
+    x = chi*(1+a**2/mag2)
+    y = eta*(1-a**2/mag2)
+    return Body(np.array([x,y]).T)
+
+def karman_trefftz(xcenter=-.1, ycenter=0, a=.1, angle_deg=10, numpoints=32):
+    """
+    Return a Karman-Trefftz foil.  This is a modified version of the Joukowski
+    foil but with a nonzero interior angle, rather than a cusp, at the trailing
+    edge.  The parameters xcenter, ycenter, a, and numpoints are the same as 
+    for joukowski_foil(), and angle_deg is the interior angle, in degrees, 
+    at the trailing edge.
+    
+    See https://en.wikipedia.org/wiki/Joukowsky_transform"""
+    angle_rad = angle_deg*np.pi/180
+    n = 2-angle_rad/np.pi
+    t = np.linspace(0,2*np.pi,numpoints)
+    ctr = xcenter + 1j*ycenter
+    r = np.linalg.norm(ctr-a)
+    zeta = ctr+r*np.exp(1j*t)
+    mag2 = np.linalg.norm(zeta)
+    z = n*((1+1/zeta)**n+(1-1/zeta)**n)/((1+1/zeta)**n-(1-1/zeta)**n)
+    x = [w.real for w in z]
+    y = [w.imag for w in z]
+    return Body(np.array([x,y]).T)
+
+def van_de_vooren_foil(semichord=1.0, thickness=0.15, angle_deg=5,
+numpoints=32):
+    """
+    Return a van de Vooren foil.
+
+    semichord = half the chord c, so c=2l
```

Comment:
Again, see comment above about a consistent way to specify chord (or half-chord).  To me, chord seems more natural for the user, but maybe semichord is more conventional?  Also, for these nice detailed comments describing the parameters, would be good to format them using the numpydoc syntax, so they look good in Sphinx, readthedocs, etc.